### ISSUE TITLE
invalid argument error

### ISSUE THREAD

**Issue by semmons99** [moranje/alfred-workflow-todoist#22]
Howdy. I am getting an error anytime I try to create a new todo. Listing items is not an issue. Here is the error console output.

```
[2017-07-31 11:03:11][input.scriptfilter] Queuing argument '(null)'
[2017-07-31 11:03:11][input.scriptfilter] Queuing argument ''
[2017-07-31 11:03:11][input.scriptfilter] Script with argument '' finished
[2017-07-31 11:03:11][STDERR: input.scriptfilter] ./shell/todo.parse.sh: line 52: INBOX          : command not found
./shell/todo.parse.sh: line 53: todoist_create_task: command not found
./shell/todo.parse.sh: line 54: : command not found
[2017-07-31 11:03:11][input.scriptfilter] {"items":[ {"uid":"title","arg":"","valid":true,"autocomplete":"","icon":"icon.png","title":"Create new task - ","subtitle":" #"}]}
[2017-07-31 11:03:12][input.scriptfilter] Queuing argument 'g'
[2017-07-31 11:03:12][input.scriptfilter] Script with argument 'g' finished
[2017-07-31 11:03:13][STDERR: input.scriptfilter] ./shell/todo.parse.sh: line 52: INBOX          : command not found
./shell/todo.parse.sh: line 53: todoist_create_task: command not found
./shell/todo.parse.sh: line 54: : command not found
[2017-07-31 11:03:13][input.scriptfilter] {"items":[ {"uid":"title","arg":"","valid":true,"autocomplete":"","icon":"icon.png","title":"Create new task - g","subtitle":" #"}]}
[2017-07-31 11:03:13][input.scriptfilter] Queuing argument 'ge'
[2017-07-31 11:03:13][input.scriptfilter] Queuing argument 'get'
[2017-07-31 11:03:13][input.scriptfilter] Script with argument 'ge' finished
[2017-07-31 11:03:13][STDERR: input.scriptfilter] ./shell/todo.parse.sh: line 52: INBOX          : command not found
./shell/todo.parse.sh: line 53: todoist_create_task: command not found
./shell/todo.parse.sh: line 54: : command not found
[2017-07-31 11:03:13][input.scriptfilter] {"items":[ {"uid":"title","arg":"","valid":true,"autocomplete":"","icon":"icon.png","title":"Create new task - ge","subtitle":" #"}]}
[2017-07-31 11:03:13][input.scriptfilter] Queuing argument 'get '
[2017-07-31 11:03:13][input.scriptfilter] Script with argument 'get' finished
[2017-07-31 11:03:13][STDERR: input.scriptfilter] ./shell/todo.parse.sh: line 52: INBOX          : command not found
./shell/todo.parse.sh: line 53: todoist_create_task: command not found
./shell/todo.parse.sh: line 54: : command not found
[2017-07-31 11:03:13][input.scriptfilter] {"items":[ {"uid":"title","arg":"","valid":true,"autocomplete":"","icon":"icon.png","title":"Create new task - get","subtitle":" #"}]}
[2017-07-31 11:03:13][input.scriptfilter] Queuing argument 'get m'
[2017-07-31 11:03:13][input.scriptfilter] Script with argument 'get ' finished
[2017-07-31 11:03:13][STDERR: input.scriptfilter] ./shell/todo.parse.sh: line 52: INBOX          : command not found
./shell/todo.parse.sh: line 53: todoist_create_task: command not found
./shell/todo.parse.sh: line 54: : command not found
[2017-07-31 11:03:13][input.scriptfilter] {"items":[ {"uid":"title","arg":"","valid":true,"autocomplete":"","icon":"icon.png","title":"Create new task - get","subtitle":" #"}]}
[2017-07-31 11:03:13][input.scriptfilter] Script with argument 'get m' finished
[2017-07-31 11:03:13][STDERR: input.scriptfilter] ./shell/todo.parse.sh: line 52: INBOX          : command not found
./shell/todo.parse.sh: line 53: todoist_create_task: command not found
./shell/todo.parse.sh: line 54: : command not found
[2017-07-31 11:03:13][input.scriptfilter] {"items":[ {"uid":"title","arg":"","valid":true,"autocomplete":"","icon":"icon.png","title":"Create new task - get m","subtitle":" #"}]}
[2017-07-31 11:03:13][input.scriptfilter] Queuing argument 'get mi'
[2017-07-31 11:03:13][input.scriptfilter] Script with argument 'get mi' finished
[2017-07-31 11:03:13][STDERR: input.scriptfilter] ./shell/todo.parse.sh: line 52: INBOX          : command not found
./shell/todo.parse.sh: line 53: todoist_create_task: command not found
./shell/todo.parse.sh: line 54: : command not found
[2017-07-31 11:03:13][input.scriptfilter] {"items":[ {"uid":"title","arg":"","valid":true,"autocomplete":"","icon":"icon.png","title":"Create new task - get mi","subtitle":" #"}]}
[2017-07-31 11:03:13][input.scriptfilter] Queuing argument 'get mil'
[2017-07-31 11:03:14][input.scriptfilter] Queuing argument 'get milk'
[2017-07-31 11:03:14][input.scriptfilter] Script with argument 'get mil' finished
[2017-07-31 11:03:14][STDERR: input.scriptfilter] ./shell/todo.parse.sh: line 52: INBOX          : command not found
./shell/todo.parse.sh: line 53: todoist_create_task: command not found
./shell/todo.parse.sh: line 54: : command not found
[2017-07-31 11:03:14][input.scriptfilter] {"items":[ {"uid":"title","arg":"","valid":true,"autocomplete":"","icon":"icon.png","title":"Create new task - get mil","subtitle":" #"}]}
[2017-07-31 11:03:14][input.scriptfilter] Script with argument 'get milk' finished
[2017-07-31 11:03:14][STDERR: input.scriptfilter] ./shell/todo.parse.sh: line 52: INBOX          : command not found
./shell/todo.parse.sh: line 53: todoist_create_task: command not found
./shell/todo.parse.sh: line 54: : command not found
[2017-07-31 11:03:14][input.scriptfilter] {"items":[ {"uid":"title","arg":"","valid":true,"autocomplete":"","icon":"icon.png","title":"Create new task - get milk","subtitle":" #"}]}
[2017-07-31 11:03:15][input.scriptfilter] Processing output of 'action.script' with arg ''
[2017-07-31 11:03:16][ERROR: action.script] module.js:487
    throw err;
    ^

Error: Cannot find module '/Users/shane/Dropbox (Personal)/Alfred.alfredpreferences/workflows/user.workflow.CD76D52B-76B1-4244-9C26-FE9AEFCC0358/index.js'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3
[2017-07-31 11:03:16][action.script] Processing output of 'utility.junction' with arg 'Error: Invalid argument value","command_type":"item_add","error_code":20,"http_code":400}},"temp_id_mapping":{},"full_sync":true,"sync_token":"J7SsbmklDw6ctwUmOhtBz7Me9c9lufgJE1PS9nltVRqRbNaVp5H7AKMLCD08fZGeutRzqHFwp6Feqe8i2NAg9XQqnsupYCMdjhu0diMw6uS4QYk'
[2017-07-31 11:03:16][utility.junction] Processing output of 'utility.filter' with arg 'Error: Invalid argument value","command_type":"item_add","error_code":20,"http_code":400}},"temp_id_mapping":{},"full_sync":true,"sync_token":"J7SsbmklDw6ctwUmOhtBz7Me9c9lufgJE1PS9nltVRqRbNaVp5H7AKMLCD08fZGeutRzqHFwp6Feqe8i2NAg9XQqnsupYCMdjhu0diMw6uS4QYk'
[2017-07-31 11:03:16][utility.filter] Processing output of 'output.notification' with arg 'Error: Invalid argument value","command_type":"item_add","error_code":20,"http_code":400}},"temp_id_mapping":{},"full_sync":true,"sync_token":"J7SsbmklDw6ctwUmOhtBz7Me9c9lufgJE1PS9nltVRqRbNaVp5H7AKMLCD08fZGeutRzqHFwp6Feqe8i2NAg9XQqnsupYCMdjhu0diMw6uS4QYk'
[2017-07-31 11:03:16][utility.junction] Processing output of 'utility.filter' with arg 'Error: Invalid argument value","command_type":"item_add","error_code":20,"http_code":400}},"temp_id_mapping":{},"full_sync":true,"sync_token":"J7SsbmklDw6ctwUmOhtBz7Me9c9lufgJE1PS9nltVRqRbNaVp5H7AKMLCD08fZGeutRzqHFwp6Feqe8i2NAg9XQqnsupYCMdjhu0diMw6uS4QYk'
```

Please let me know if there is anything I can do to help debug this.

**Comment by chrisramakers** [comment]
Same here, adding tasks doesn't seem to work ... api gives this output

`Error: Invalid argument value","command_type":"item_add","error_code":20,"http_code":400}},"temp_id_mapping":{},"full_sync":true,"sync_token":"tKK_Nh8YVTGKdj-LWsl4iKAbzGkM2pVITpmJg9mOUNWERffnN4sd_oPY6g1gXjdg7RVVt84U6lAkkm5runGkQ4CbsUk7aejdp8fRZBDNNYFi`

**Comment by dkpalmer** [comment]
Have run into this as well. Looks like it's the same issue as https://www.alfredforum.com/topic/8554-todoist-workflow/

**Comment by moranje** [comment]
Yeah I'm sorry for the delay I'm working on an update that includes a test
suite and a switch from bash to mostly js. Stay with me.

-- 
Martien Oranje

Op 8 augustus 2017 bij 14:54:11, Danny Palmer (notifications@github.com)
schreef:

Have run into this as well. Looks like it's the same issue as
https://www.alfredforum.com/topic/8554-todoist-workflow/

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/moranje/alfred-workflow-todoist/issues/22#issuecomment-320947048>,
or mute the thread

**Comment by semmons99** [comment]
@moranje is there anything we can help with? I am pretty well versed in js.

**Comment by moranje** [comment]
It's coming along, but you'll have to forgive the pace in favor of a steady release. This one will have a test suite to go along with it and is mostly written in js (well typescript). Hang on!

**Comment by moranje** [comment]
I want to keep you in the loop. I have finished rewriting everything in typescript. I need to write a bunch of tests to make sure that everything works as expected. should be version 4.0.0. I can work on including more functionality if that's released.

**Comment by semmons99** [comment]
![aww yeah](https://media.giphy.com/media/3oz8xRF0v9WMAUVLNK/giphy.gif)

**Comment by NeverMoes** [comment]
Waiting for the fix, thanks a lot.

**Comment by moranje** [comment]
The update is live (as a sort of pre release). Source code will come shortly but you can download the workflow on github or using `t:update`.

Martien

**Comment by dkpalmer** [comment]
Just gave it a try. Todo listing looks great. However, I'm still unable to create a todo w/ v4.0.0:

<img width="557" alt="screen shot 2017-09-15 at 5 30 50 am" src="https://user-images.githubusercontent.com/2394698/30482257-116a8b12-99d7-11e7-830b-b10c728528e7.png">

```
[2017-09-15 05:26:20][input.scriptfilter] Script with argument '(null)' finished

[2017-09-15 05:26:20][input.scriptfilter] {"items":[{"title":"Create new task - Test","uid":"todoist_create_task","arg":"{\"labels\":[],\"priority\":null,\"project\":\"Inbox\",\"date\":\"\",\"task\":\"Test\"}","type":"default","valid":true,"autocomplete":"Test","icon":"icon.png","subtitle":"INBOX          "},{"title":"Set project","uid”:”<my projid>","arg":"","type":"default","valid":false,"autocomplete":"Test #","icon":{"path":"project.png"},"subtitle":"Store task in a project, ex. #Work"},{"title":"Add label","uid”:”<the uid>","arg":"","type":"default","valid":false,"autocomplete":"Test @","icon":{"path":"label.png"},"subtitle":"Add a label to this task, ex. @15min"},{"title":"Set priority","uid”:”<the uid>","arg":"","type":"default","valid":false,"autocomplete":"Test !!","icon":{"path":"priority.png"},"subtitle":"Set task priority, ex. !!3"},{"title":"Set date","uid”:”<uid>","arg":"","type":"default","valid":false,"autocomplete":"Test , ","icon":{"path":"date.png"},"subtitle":"Set a date using a todoist date string, ex. today"}]}

[2017-09-15 05:26:20][input.scriptfilter] Processing output of 'action.script' with arg '{"labels":[],"priority":null,"project":"Inbox","date":"","task":"Test"}'

[2017-09-15 05:26:21][action.script] Processing output of 'utility.junction' with arg 'Error: {}

[2017-09-15 05:26:21][utility.junction] Processing output of 'utility.filter' with arg 'Error: {}

[2017-09-15 05:26:21][utility.filter] Processing output of 'output.notification' with arg 'Error: {}

[2017-09-15 05:26:21][utility.junction] Processing output of 'utility.filter' with arg 'Error: {}

```

**Comment by moranje** [comment]
You have entered the api key in `t:token`?

-- 
Martien Oranje

Op 15 september 2017 bij 14:32:09, Danny Palmer (notifications@github.com)
schreef:

Just gave it a try. Todo listing looks great. However, I'm still unable to
create a todo w/ v4.0.0:

[image: screen shot 2017-09-15 at 5 30 50 am]
<https://user-images.githubusercontent.com/2394698/30482257-116a8b12-99d7-11e7-830b-b10c728528e7.png>

[2017-09-15 05:26:20][input.scriptfilter] Script with argument '(null)' finished

[2017-09-15 05:26:20][input.scriptfilter] {"items":[{"title":"Create
new task - Test","uid":"todoist_create_task","arg":"{\"labels\":[],\"priority\":null,\"project\":\"Inbox\",\"date\":\"\",\"task\":\"Test\"}","type":"default","valid":true,"autocomplete":"Test","icon":"icon.png","subtitle":"INBOX
         "},{"title":"Set project","uid”:”<my
projid>","arg":"","type":"default","valid":false,"autocomplete":"Test
#","icon":{"path":"project.png"},"subtitle":"Store task in a project,
ex. #Work"},{"title":"Add label","uid”:”<the
uid>","arg":"","type":"default","valid":false,"autocomplete":"Test
@","icon":{"path":"label.png"},"subtitle":"Add a label to this task,
ex. @15min"},{"title":"Set priority","uid”:”<the
uid>","arg":"","type":"default","valid":false,"autocomplete":"Test
!!","icon":{"path":"priority.png"},"subtitle":"Set task priority, ex.
!!3"},{"title":"Set
date","uid”:”<uid>","arg":"","type":"default","valid":false,"autocomplete":"Test
, ","icon":{"path":"date.png"},"subtitle":"Set a date using a todoist
date string, ex. today"}]}

[2017-09-15 05:26:20][input.scriptfilter] Processing output of
'action.script' with arg
'{"labels":[],"priority":null,"project":"Inbox","date":"","task":"Test"}'

[2017-09-15 05:26:21][action.script] Processing output of
'utility.junction' with arg 'Error: {}

[2017-09-15 05:26:21][utility.junction] Processing output of
'utility.filter' with arg 'Error: {}

[2017-09-15 05:26:21][utility.filter] Processing output of
'output.notification' with arg 'Error: {}

[2017-09-15 05:26:21][utility.junction] Processing output of
'utility.filter' with arg 'Error: {}


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/moranje/alfred-workflow-todoist/issues/22#issuecomment-329769342>,
or mute the thread

**Comment by semmons99** [comment]
same for me even after storing the token

**Comment by moranje** [comment]
No errors either?

-- 
Martien Oranje

Op 15 september 2017 bij 14:56:01, Shane Emmons (notifications@github.com)
schreef:

same for me even after storing the token

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/moranje/alfred-workflow-todoist/issues/22#issuecomment-329774923>,
or mute the thread

**Comment by semmons99** [comment]
Here is the debug info

```
[2017-09-15 09:08:17][input.scriptfilter] Processing output of 'action.script' with arg '{"labels":[],"priority":null,"project":"work","date":"","task":"tod test"}'
[2017-09-15 09:08:18][action.script] Processing output of 'utility.junction' with arg 'Error: {}
'
[2017-09-15 09:08:18][utility.junction] Processing output of 'utility.filter' with arg 'Error: {}
'
[2017-09-15 09:08:18][utility.filter] Processing output of 'output.notification' with arg 'Error: {}
'
[2017-09-15 09:08:18][utility.junction] Processing output of 'utility.filter' with arg 'Error: {}
'
```

**Comment by dkpalmer** [comment]
Yep, token entered (confirmed I'm able to successfully list tasks via `todos`)

**Comment by moranje** [comment]
Try a `t:update`, should be fixed now.

-- 
Martien Oranje

Op 15 september 2017 bij 15:11:24, Danny Palmer (notifications@github.com)
schreef:

"{\"labels\":[],\"priority\":null,\"project\":\"Inbox\",\"date\":\"\",\"task\":\"Test\"}","type":"default","valid":true,"autocomplete":"Test","icon":"icon.png","subtitle":"INBOX
         "},{"title":"Set project","uid”:”<my
projid>","arg":"","type":"default","valid":false,"autocomplete":"Test
#","icon":{"path":"project.png"},"subtitle":"Store task in a project,
ex. #Work"},{"title":"Add label","uid”:”<the
uid>","arg":"","type":"default","valid":false,"autocomplete":"Test
@","icon":{"path":"label.png"},"subtitle":"Add a label to this task,
ex. @15min"},{"title":"Set priority","uid”:”<the
uid>","arg":"","type":"default","valid":false,"autocomplete":"Test
!!","icon":{"path":"priority.png"},"subtitle":"Set task priority, ex.
!!3"},{"title":"Set
date","uid”:”<uid>","arg":"","type":"default","valid":false,"autocomplete":"Test
, ","icon":{"path":"date.png"},"subtitle":"Set a date using a todoist
date string, ex. today"}]}

**Comment by seeARMS** [comment]
Same issues as above, but running a t:update first gives a `permission denied` error, then after running `chmod a+rwx ./shell/t.update.sh` it now says no updates are available

**Comment by moranje** [comment]
Try again, I ave included a fix for the permission denied problem with
script files.

-- 
Martien Oranje

Op 15 september 2017 bij 16:36:34, Colin Armstrong (notifications@github.com)
schreef:

Same issues as above, but running a t:update first gives a permission denied
error, then after running chmod a+rwx ./shell/t.update.sh it now says no
updates are available

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/moranje/alfred-workflow-todoist/issues/22#issuecomment-329800854>,
or mute the thread

**Comment by dkpalmer** [comment]
Same issue as @seeARMS. Unable to install. I tried uninstalling my workflow and same result with download link on master.

**Comment by semmons99** [comment]
4.0.2 does not install for me either

**Comment by moranje** [comment]
Thanks for your patience. I tried to create the binairy through script but
that seems to be failing so I reverted back creating a manual export.
Download should now work.

-- 
Martien Oranje

Op 15 september 2017 bij 20:09:12, Shane Emmons (notifications@github.com)
schreef:

4.0.2 does not install for me either

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/moranje/alfred-workflow-todoist/issues/22#issuecomment-329857518>,
or mute the thread
